### DISCUSSION TITLE
Do we need both `run_if` and `in_set`?

### DISCUSSION THREAD

**Discussion by torsteingrindvik** [bevyengine/bevy#10863] [Ideas]
In my code I have lots of systems which end in

```rust
.run_if(in_state(STATE))
.in_set(SET),
```

and `in_set` just feels like another `run_if` condition.

In the spirit of reducing API surface, would it be better to be able to just

```rust
.run_if(in_state(STATE))
.run_if(in_set(SET)),
```
?

**Comment by alice-i-cecile** [comment]
System sets are a more fundamental primitive than run conditions: they can be nested, ordered, configured and so on.

In code like your first example, I tend to try and attach the run condition to the set itself.

**Comment by torsteingrindvik** [comment]
Ok, thanks for the feedback. I'll try to apply your method to my code and see if that works out.
